### DISCUSSION TITLE
Примеры реализации + используемые компоненты

### DISCUSSION THREAD

**Discussion by saippuakauppias** [vvip-68/LedPanelWiFi#52] [Похвастушки]
Решил на своем примере рассказать что в итоге собрал, чтобы коллективно постепенно собирать используемые компоненты, да и просто похвастаться как вышло :)

Немного: предыстории – проект этот помню ещё с загрузки видео Александром на ютубе, но несколько лет подряд под новый год не было времени и сил его попробовать «на живую», а в этот год совпали звёзды и я решился. Вначале хотел как и все – собрать всё на окне, но т.к. мой уровень пайки и познаний вокруг МК и компонентов скорее как у начинающего – было немного страшно оставлять такое рядом со шторой 🤣 Поэтому в какой-то момент я осознал, что можно тоже самое сделать и на заборе на улице – благо места там предостаточно, а риск замыкания/пожара там не такой страшный 🤣 
Т.к. монтировать ленту на сам забор зимой так себе приключение, поэтому придумал что вначале сделаю деревянный каркас, закреплю всю ленту дома (и опережая историю – это было отличная идея), а уже потом вынесу всё на улицу и просто подключу к питанию.

И вот тут уже можно начать. Решил что вся гирлянда будет размером 4 метра шириной и 1.2 метра высотой (больше в высоту можно, но есть риск что выпадет много снега и сугробом закроет часть). Поэтому закупил в леруа 13 брусков 20х30 длинной 2м. Из них решил собрать сам каркас, точные размеры не дам, т.к. подгонял ещё по факту, но основывался вот на этой схеме:
![IMG_6797](https://github.com/vvip-68/LedPanelWiFi/assets/945306/003af7ac-b76c-46f7-b6ee-cfa5c10c49fa)
Купил [краску по дереву](https://www.ozon.ru/product/kraska-dlya-dereva-akrilovaya-polumatovaya-husky-siberian-0-9l-grafitovyy-ral-7024-s-naturalnym-893982160/?oos_search=false) в цвет забора (чтоб не сильно всё выделялось) и покрасил их прямо дома:
![IMG_6647](https://github.com/vvip-68/LedPanelWiFi/assets/945306/527a8098-5fe1-4af9-9423-4bcb4a2785b5)

Всё это делалось пока я ожидал ленту и БП. [Ленту решил заказать](https://aliexpress.ru/item/1005005940012280.html?spm=a2g2w.orderdetail.0.0.68364aa641KITG&sku_id=12000035163628063) бабиной 100м с расстоянием между диодами 5 см, где фактически 2000 диодов (_в итоге когда она пришла – я осознал, что надо было брать с тёмным проводом, но уже было поздно_). А БП взял сразу с [огромным запасом в 60А](https://aliexpress.ru/item/32670505021.html?spm=a2g2w.orderdetail.0.0.78c94aa696EFh0&sku_id=67475760404), потому что не понимал фактическое потребление (_когда собирал в спешке всё – так и не замерил сколько потребляется, но явно сильно меньше чем сам БП_).
![IMG_6630](https://github.com/vvip-68/LedPanelWiFi/assets/945306/edb274aa-ab2a-4ba9-88d4-551ec4cdd7fa)
![IMG_6663](https://github.com/vvip-68/LedPanelWiFi/assets/945306/8aae9a51-b77c-4386-b7d3-e5152ceeb024)

В качестве МК купил ESP32 на чипе CH9102X, очень долго пытался его прошить теми кабелями, что у меня были (даже попросил у соседа всё что есть), но оказалось что они все – не data-кабели. Поэтому купил отдельно [data-кабель](https://www.ozon.ru/product/kabel-dlya-zaryadki-i-peredachi-dannyh-exployd-data-cable-usb-micro-usb-1121407112/):
![IMG_6580](https://github.com/vvip-68/LedPanelWiFi/assets/945306/e9845891-d05a-4251-9451-a447c0f7ee05)
![IMG_6623](https://github.com/vvip-68/LedPanelWiFi/assets/945306/1ce5bc96-2f0d-46ed-9882-1e9c768a5076)

В закромах завалялся старый набор от амперки, где был [SD-shield](https://amperka.ru/product/troyka-sd) – решил использовать его, докупив [micro-SD карту на 4ГБ](https://www.ozon.ru/product/karta-pamyati-perfeo-ultra-speed-4-gb-pf4gmcsh10aes-662434308/).

В итоге всё прошил, протестил работоспособность на ленте (просто приклеив её к полу на скотч, размером 10х14 диодов) – всё заработало практически с первого раза. Счастью моему тогда не было предела :)

После этого приступил к сборке каркаса. Примерно за пол дня собрал его (отпиливая лишнее, где обсчитался с размерами), а потом начал монтировать ленту на него. Тут стоит отметить важную идею (без которой ничего бы не получилось): ленту я крепил через строительный степлер, но чтобы от удара степлера контакты не рвались – я взял ВВГ провод, снял с него верхнюю изоляцию и порезал на кусочки по 10-15мм, которые клал сверху на ленту и фактически этим прижимал сильнее саму ленту к каркасу + исключал перебитие контактов. В итоге за все 216 соединений – нигде ничего не перебилось и всё заработало с первого раза по итогу :)
![IMG_6738](https://github.com/vvip-68/LedPanelWiFi/assets/945306/dec02bb0-ab3e-446b-a48a-f6e1b83d057c)

Так как ленту делал "зигзагом" – конечно же на поворотах не хватало длины между диодами, чтобы они шли с идеально ровным расстоянием между друг другом в 5 см. Поэтому каждый изгиб – резал, и впаивал перемычки для соединения всех контактов. К слову, тут же сразу же стоит отметить – что подключал +5V/-5V каждый 2й ряд, чтобы лента светилась равномерно. Вся гирлянда монтировалась из расчета: **24 ряда по 80 диодов** в каждом (т.е. питание каждые 160 диодов). 

После окончания монтажа – проверил с бредбордом что всё работает ожидаемо и начал распаивать МК + SD-shield на [печатной макетной плате](https://www.ozon.ru/product/pechatnaya-maketnaya-plata-3-x-7-sm-dvuhstoronnyaya-dlya-payki-5-shtuk-917242107/). Ставил всё само собой по схеме: конденсатор на питании, резистор перед лентой, все пины согласно распиновке. Вышло не очень красиво (потому что куча перемычек), но т.к. паял последний раз лет 7 назад – вполне работоспособно.
![IMG_6741](https://github.com/vvip-68/LedPanelWiFi/assets/945306/7351708f-3207-40c1-8ba5-3ecf20c2b350)
![IMG_6744](https://github.com/vvip-68/LedPanelWiFi/assets/945306/bf64872e-8e17-4e2f-b2aa-eadb36b784c9)

Так как сам МК будет находиться на улице – нужно было защитить его корпусом. Погуглил стоимость корпусов примерно в размер и офигел. Поэтому по славной традиции Гайвера – решил не париться и собрать всё просто в сантехе 🤣 Купил муфту на 50 и две заглушки. Что же может быть герметичнее чем это решение?)
![IMG_6748](https://github.com/vvip-68/LedPanelWiFi/assets/945306/b97b9819-a20d-4d19-9691-8d2d85e9ffa5)

БП тоже, конечно же размещается на улице, поэтому тут тоже не вышло без изврата. Купил обычный пластиковый контейнер с крышкой в том же леруа чуть большего размера, поставил сам БП на П-образные "ножки" из din-рейки (просто на всякий случай, чтоб он не был на дне), подключил всё окончательно для теста дома – и всё также работало. Ну не чудеса ли это?))

Потом попросил помощи друга-соседа, чтобы вынести каркас с лентой (он легкий, но очень громоздкий), закрепили его на забор на стяжки, подключил питание, замотал крышку контейнера у БП на тканевый серебристый скотч в несколько раз (чтобы исключить попадание влаги, т.к. он будет жить в сугробе), подключил БП через розетку-таймер (механический) к 220В, настроив время запуска с 16ч до 23ч и всё ожило!

Итоговый результат (видео): https://youtu.be/NwCYKiYOzOs

<img width="667" alt="Снимок экрана 2023-12-31 в 14 26 48" src="https://github.com/vvip-68/LedPanelWiFi/assets/945306/dd59448c-5181-4a08-bf13-dea05f6c86e3">

**Comment by Andriejus** [comment]
Nice. Do you have more photos of your frame for LEDs? What kind of diffusion are you using? How you made for water resistance? Interesting to see how it looks at daytime.

**Comment by shkolin1973** [comment]
Вот мой результат!  https://youtube.com/shorts/_jqKmQr7lxw?feature=share
Матрица 34х17. Использовал уличную гирлянду с шагом диодов 10см с прозрачным проводом. На стекло клеил обычным прозрачным скотчем (верх, низ и в двух местах по середине столбцов). Контроллер ESP-32. Гирлянда проложена вверх-вниз, слева на право. 4 стекла. Переходы между стеклами на разъемах. Подгадал, чтобы на одной открывной створке заход на створку и переход на следующую створку были внизу. На проветривание можно открывать спокойно. А если нужно совсем распахнуть, отключаем один разъем и открываем. Проложен дополнительный провод питания по верху рамы 2*0,75мм2  и от него на гирлянду в 4-х местах подано дополнительное питание. Без этого работать не будет. Потребление гирлянды не более 5В/1А на 100диодов.
Если есть вопросы отвечу.
Автору (авторам) проекта большое спасибо за работу! Это очень красиво! Показал видео коллегам по работе. Никто ничего подобного у нас в городе не видел, и я кстати тоже. Люди вещают гирлянды на окна, но ничего похожего нет.
Я из Новосибирска. Очень интересно узнать кто из каких городов повторили проект.
И ещё, на работе в офисе ёлку украсили такой же гирляндой, матрица 32х16, тоже красиво получилось!
Спасибо за труды!

**Reply by polyg1-games** [reply]
Требуется доработка кода под esp32? 
Так как автор приводит пример подключения только esp8266.

**Reply by vvip-68** [reply]
@polyg1-games Не требуется. Автор приводит примеры настройки подключения и к esp8266 и к esp32
